### DISCUSSION TITLE
Subpath is ignored on js resource requests

### DISCUSSION THREAD

**Discussion by vinokurig** [quarkusio/quarkus#40241] [Q&A]
Hello, I am trying to launch quarkus dev-ui application on an endpoint with subpath: `https://host.com/subpath`. I see the welcome page from the subpath url: `https://host.com/subpath/q/dev-ui/welcome` but other resources are requested by the url without the subpath, e.g. `https://host.com/q/_static/es-module-shims/1.8.3/dist/es-module-shims.js`:
![Screenshot 2024-04-24 at 13 12 21(1)](https://github.com/quarkusio/quarkus/assets/7668752/b49ea0ab-8a96-4c30-9235-08a57630adf8)

If I use the `quarkus.http.root-path` property set to `subpath` the welcome page is available by the url with an additional subpath: `https://host.com/subpath/subpath/q/dev-ui/welcome`, however the js resources are available by the url with a single subpath: `https://host.com/subpath/q/_static/es-module-shims/1.8.3/dist/es-module-shims.js`. This causes the resource not found page:

![screenshot-che-dogfooding apps che-dev x6e0 p1 openshiftapps com-2024 04 24-13_49_26](https://github.com/quarkusio/quarkus/assets/7668752/6d6975bb-612b-4d6b-91a2-73e0f9e5a116)

**Comment by phillip-kruger** [comment]
Are you using localhost ? Dev UI only works in localhost. If you are using localhost, please share a reproducer I can look at.

**Comment by vinokurig** [comment]
@phillip-kruger 
I am using [Eclipse Che](https://eclipse.dev/che/) which exposes localhost to a url with subpath (`https://host.com/subpath`). Let me know If you want me to elaborate more on how to setup an environment to start the quarkus dev-ui application with Eclipse Che.

**Reply by phillip-kruger** [reply]
Yes please

**Comment by vinokurig** [comment]
@phillip-kruger 
You can use RedHat devspaces, but first you will need to create an account.
1. Start a workspcae from the given [url](https://workspaces.openshift.com/f?url=https://github.com/vinokurig/code-with-quarkus.git)
2. Wait until workspace starts and go to menu -> Terminal -> Run Task...
![screenshot-devspaces apps sandbox-m4 g2pi p1 openshiftapps com-2024 05 09-14_53_17](https://github.com/quarkusio/quarkus/assets/7668752/e1777bd5-80d6-41bb-bf0a-ef556db80842)
3. Select `devfile` from the dropdown:
![screenshot-devspaces apps sandbox-m4 g2pi p1 openshiftapps com-2024 05 09-14_54_04](https://github.com/quarkusio/quarkus/assets/7668752/80b5b834-207f-4b07-97a5-f865df59f93e)
4. Select `devfile: dev-run` from the dropdown: 
![screenshot-devspaces apps sandbox-m4 g2pi p1 openshiftapps com-2024 05 09-14_55_03](https://github.com/quarkusio/quarkus/assets/7668752/c4626f2e-6d39-4218-be9f-dcb6b6b401f1)
This will open a terminal and execute the next command: `mvn quarkus:dev -Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager`
5. Wait a notification about port 8080 is ready, click the `Open In New Tab` button:
![screenshot-devspaces apps sandbox-m4 g2pi p1 openshiftapps com-2024 05 09-14_59_00](https://github.com/quarkusio/quarkus/assets/7668752/3e68ac42-42e2-46c0-af16-d26d5b1c9322)
6. In the new tab you will see an empty page, please add the correct subpath to the url, so the url should look like: `https://devspaces.apps.sandbox-m4.g2pi.p1.openshiftapps.com/<username>/java-quarkus/8080/q/dev-ui/welcome` and load the new url.

**Comment by phillip-kruger** [comment]
@stuartwdouglas @sberyozkin - This also did not work on the old Dev UI. Any suggestions ?

**Comment by stuartwdouglas** [comment]
I think this is a che issue, I could not reproduce it locally.

**Reply by vinokurig** [reply]
Hello, did you use subpath while reproducing?

**Reply by stuartwdouglas** [reply]
Yes, do you have a way to reproduce it outside of che?

**Reply by vinokurig** [reply]
I do not have another way to reproduce the issue. Could you please describe how did you start the dev-ui locally with a subpath url? That would help a lot to investigate why it does not work with Eclipse Che.

**Reply by vinokurig** [reply]
I am curious how did you map your localhost to a url with subpath?
